priam/src/main/java/com/netflix/priam/backup/SnapshotBackup.java

```diff
@@ -99,6 +103,21 @@ public void execute() throws Exception {
             sleeper.sleep(WAIT_TIME_MS);
         }
 
+        // Do not allow more than one snapshot to run at the same time. This is possible as this
+        // happens on CRON.
+        if (!lock.tryLock()) {
+            logger.error("Snapshot Operation is already running! Try again later.");
```

Comment:
nit: maybe a warn, since this is not necessary caused by application, but rather due to crons?